drop_merge_sort.hpp

```diff
@@ -126,12 +126,13 @@ namespace detail {
                     ++read;
                     ++num_dropped_in_row;
                 } else {
-                    size_t trunc_to_length = dropped.size() - num_dropped_in_row;
                     for (int i = 0; i < num_dropped_in_row; ++i) {
                         --read;
                         *read = std::move(*(dropped.end() - (i+1)));
                     }
-                    dropped.resize(trunc_to_length);
+                    for (int i = 0; i < num_dropped_in_row; ++i) {
```

Comment:
This can most likely be merged with the loop above:

                    for (int i = 0; i < num_dropped_in_row; ++i) {
                        --read;
                        *read = std::move(*(dropped.end() - 1));
                        dropped.pop_back();
                    }